blog/2024-02-27-deprecate-bind-hass-and-hass-components.md

```diff
@@ -0,0 +1,90 @@
+---
+author: Jan-Philipp Benecke
+authorURL: https://github.com/jpbede
+authorImageURL: https://avatars.githubusercontent.com/u/3989428?s=96&v=4
+title: "Deprecate use of @bind_hass and hass.components"
+---
+
+As of Home Assistant 2024.3, we deprecate the use of the `@bind_hass` decorator 
+and thus also the use of `hass.components`.
+With the release of Home Assistant 2024.3, we are starting to adjust and remove usage in Home Assistant Core.
+Authors of custom integrations will need to adjust their code to avoid breakage.
+
+Starting from Home Assistant 2024.6, the `@bind_hass` decorator and 
+`hass.components` will be removed and will no longer work.
+
+## Use of `@bind_hass` decorator
+
+Integrations that use the `@bind_hass` decorator should be updated to remove them and pass the `hass` object as first parameter to the function instead:
+
+### New example
+
+```python
+from homeassistant.core import HomeAssistant
+from homeassistant.components.persistent_notification import async_create
+
+def create_notification(hass: HomeAssistant, message: str):
+    """Create a notification."""
+    async_create(
+        hass,
+        message, 
+        title='Important notification'
+    )
+
+async def async_setup(hass: HomeAssistant, config):
+    """Set up the component."""
+    create_notification(hass, "You're already using the latest version!")
+```
+
+### Old example
+
+```python
+from homeassistant.core import HomeAssistant
+from homeassistant.loader import bind_hass
+from homeassistant.components.persistent_notification import async_create
+
+@bind_hass
+def create_notification(hass: HomeAssistant, message: str):
+    """Create a notification."""
+    async_create(
+        hass,
+        message, 
+        title='Important notification'
+    )
+
+async def async_setup(hass: HomeAssistant, config):
+    """Set up the component."""
+    create_notification("You're already using the latest version!")
+```
+
+## Use of `hass.components` 
+
+Integrations that use `hass.components` should be updated to import the functions and classes directly from the integration package and pass the `hass` object as first parameter:
```

Comment:
Should this post also explain that you need to add the imported components to the manifest?

Reply:
Did this based on https://github.com/home-assistant/core/pull/111494#discussion_r1504474069.

Reply:
Yes, it's good to mention dependencies if importing from other integrations.

Reply:
Will adjust that, thanks